lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -354,10 +355,37 @@ void FIRRTLTypesLowering::visitDecl(MemOp op) {
   opsToRemove.push_back(op);
 }
 
+/// Lower a reg op with a bundle to multiple non-bundled regs.
+void FIRRTLTypesLowering::visitDecl(RegOp op) {
+  Value result = op.result();
+
+  // Attempt to get the bundle types, potentially unwrapping an outer flip type
+  // that wraps the whole bundle.
+  BundleType resultType = getCanonicalBundleType(result.getType());
+
+  // If the reg is not a bundle, there is nothing to do.
+  if (!resultType)
+    return;
+
+  SmallVector<FlatBundleFieldEntry, 8> fieldTypes;
+  flattenBundleTypes(resultType, "", false, fieldTypes);
+
+  // Loop over the leaf aggregates.
+  for (auto field : fieldTypes) {
+    setBundleLowering(
+        result, StringRef(field.suffix).drop_front(1),
+        builder->create<RegOp>(field.getPortType(), op.clockVal()));
```

Comment:
Registers can have a name, we should create one based on the existing register's name + field name.

Reply:
Updated the name: "register.name + field.suffix"